bitbots_hcm/src/bitbots_hcm/fall_checker.py

```diff
@@ -1,112 +1,71 @@
-        # rospy.set_param("hcm/falling_threshold_orientation_left_right", self.falling_threshold_orientation_left_right)
+class FallChecker(BaseEstimator):
 
-        # Grenzwerte an Untergrund anpassen
-        self.falling_threshold_front *= self.ground_coefficient
-        self.falling_threshold_side *= self.ground_coefficient
+    def __init__(self, thresh_gyro_front=rospy.get_param("hcm/threshold_gyro_y_front"),
+                 thresh_gyro_side=rospy.get_param("hcm/threshold_gyro_x_side"),
+                 thresh_orient_front=math.radians(rospy.get_param("hcm/falling_threshold_orientation_front_back")),
+                 thresh_orient_side=math.radians(rospy.get_param("hcm/falling_threshold_orientation_left_right"))):
 
-        self.FRONT = "FRONT"
-        self.BACK = "BACK"
-        self.LEFT = "LEFT"
-        self.RIGHT = "RIGHT"
-        self.SIDE = "SIDE"
+        self.thresh_gyro_front = thresh_gyro_front
+        self.thresh_gyro_side = thresh_gyro_side
+        self.thresh_orient_front = thresh_orient_front
+        self.thresh_orient_side = thresh_orient_side
 
-    def update_reconfigurable_values(self, config, level):
-        # Dynamic Reconfigure
-        self.falling_threshold_front = config["threshold_gyro_y_front"]
-        self.falling_threshold_side = config["threshold_gyro_x_side"]
-        self.falling_threshold_orientation_front_back = math.radians(config["falling_threshold_orientation_front_back"])
-        self.falling_threshold_orientation_left_right = math.radians(config["falling_threshold_orientation_left_right"])
-        return config
+        self.STABLE = 0
+        self.FRONT = 1
+        self.BACK = 2
+        self.LEFT = 3
+        self.RIGHT = 4
 
-    def check_falling(self, not_much_smoothed_gyro, quaternion):
+    def check_falling(self, not_much_smoothed_gyro, euler):
         """Checks if the robot is currently falling and in which direction. """
         # Checks if robot is still
-        if all(abs(n) < 0.1 for n in not_much_smoothed_gyro):
-            return
+        bools = [abs(n) < 0.1 for n in not_much_smoothed_gyro]
+        if bools[0] and bools[1] and bools[2]:
+            return self.STABLE
 
-        # converting the Quaternion into Euler angles for better understanding
-        euler = tf.transformations.euler_from_quaternion(quaternion)
-        if self.falling_threshold_front == 0 or \
-                self.falling_threshold_side == 0 or \
-                self.falling_threshold_orientation_front_back == 0 or \
-                self.falling_threshold_orientation_left_right == 0:
-            return
         # setting the fall quantification function
         x_fall_quantification = self.calc_fall_quantification(
-            self.falling_threshold_orientation_left_right,
-            self.falling_threshold_front,
-            euler[0] - self.pitch_offset,
+            self.thresh_orient_side,
+            self.thresh_gyro_front,
```

Comment:
These seem switched. This is no error of this PR, but I think the `x_quantification` should use `thresh_orient_side` and `thresh_gyro_side`.